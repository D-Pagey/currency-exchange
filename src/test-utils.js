import React from 'react';
import { render } from '@testing-library/react';
import { AccountsContext, initialAccounts } from './components/ProviderAccounts';

const customRender = (ui, accountsContextValue = {}) => {
    return render(
        <AccountsContext.Provider
            value={{
                accounts: initialAccounts,
                setAccounts: () => null,
                ...accountsContextValue,
            }}
        >
            {ui}
        </AccountsContext.Provider>,
    );
};

export * from '@testing-library/react';

export { customRender as render };
