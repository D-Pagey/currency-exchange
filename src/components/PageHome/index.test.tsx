import React from 'react';
import { render } from '../../test-utils';
import { PageHome } from '.';

const context = {
    accounts: {
        GBP: 50,
        USD: 50,
        EUR: 50,
    },
};

describe('PageHome component', () => {
    it('should render', () => {
        const { container } = render(<PageHome />, context);
        expect(container.firstChild).toMatchSnapshot();
    });
});
