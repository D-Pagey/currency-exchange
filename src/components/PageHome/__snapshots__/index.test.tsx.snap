// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHome component should render 1`] = `
<div>
  <h2
    class="sc-AxjAm hDBDAG"
  >
    Your Account Balances:
  </h2>
  <ul
    class="sc-AxirZ diVtBJ"
  >
    <li
      class="sc-AxiKw bMLjCv"
    >
      EUR
      : 
      €
      50
    </li>
    <li
      class="sc-AxiKw bMLjCv"
    >
      GBP
      : 
      £
      50
    </li>
    <li
      class="sc-AxiKw bMLjCv"
    >
      USD
      : 
      $
      50
    </li>
  </ul>
</div>
`;
